Mathlib/NumberTheory/ModularForms/JacobiTheta/TwoVariable.lean

```diff
@@ -0,0 +1,136 @@
+/-
+Copyright (c) 2023 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import Mathlib.Analysis.SpecialFunctions.Gaussian
+import Mathlib.Analysis.Complex.LocallyUniformLimit
+
+/-!
+# The two-variable Jacobi theta function
+
+This file defines the two-variable Jacobi theta function
+
+$$\theta(z, \tau) = \sum_{n \in \mathbb{Z}} \exp (2 i \pi n z + i \pi n ^ 2 \tau),$$
+
+and proves the functional equation relating the values at `(z, τ)` and `(z / τ, -1 / τ)`,
+using Poisson's summation formula. We also show holomorphy (in both variables individually,
+not jointly).
+-/
+
+open Complex Real Asymptotics Filter Topology
+
+open scoped Real BigOperators
+
+/-- The two-variable Jacobi theta function,
+`θ z τ = ∑' (n : ℤ), cexp (2 * π * I * n * z +  π * I * n ^ 2 * τ)`.
+
+The sum is only convergent for `0 < im τ`; we are implictly extending it by 0 for other values of
+`τ`. -/
+noncomputable def jacobiTheta₂ (z τ : ℂ) : ℂ :=
+  ∑' n : ℤ, cexp (2 * π * I * ↑n * z +  π * I * n ^ 2 * τ)
+
+/-- A uniform bound for the summands in the Jacobi theta function on compact subsets. -/
+lemma jacobiTheta₂_term_bound {S T : ℝ} (hT : 0 < T) {z τ : ℂ}
+    (hz : |im z| ≤ S) (hτ : T ≤ im τ) (n : ℤ) :
+    ‖cexp (2 * π * I * n * z + π * I * n ^ 2 * τ)‖ ≤ Real.exp (-π * (T * n ^ 2 - 2 * S * |n|)) := by
+  rw [Complex.norm_eq_abs, Complex.abs_exp, (by push_cast; ring :
+    2 * π * I * n * z + π * I * n ^ 2 * τ = (π * (2 * n) : ℝ) * z * I + (π * n ^ 2 : ℝ) * τ * I),
+    add_re, mul_I_re, im_ofReal_mul, mul_I_re, im_ofReal_mul, Real.exp_le_exp, neg_mul, ← neg_add,
+    neg_le_neg_iff, mul_assoc π, mul_assoc π, ← mul_add, mul_le_mul_left pi_pos, add_comm,
+    mul_comm T]
+  refine add_le_add (mul_le_mul le_rfl hτ hT.le (sq_nonneg _)) ?_
+  rw [← mul_neg, mul_assoc, mul_assoc, mul_le_mul_left two_pos, mul_comm, neg_mul, ← mul_neg]
+  refine le_trans ?_ (neg_abs_le_self _)
+  rw [mul_neg, neg_le_neg_iff, abs_mul, Int.cast_abs]
+  exact mul_le_mul_of_nonneg_left hz (abs_nonneg _)
+
+lemma summable_jacobiTheta₂_term_bound (S : ℝ) {T : ℝ} (hT : 0 < T) :
+    Summable (fun n : ℤ ↦ Real.exp (-π * (T * n ^ 2 - 2 * S * |n|))) := by
+  suffices : Summable (fun n : ℕ ↦ Real.exp (-π * (T * n ^ 2 - 2 * S * n)))
+  · apply summable_int_of_summable_nat <;>
+    simpa only [Int.cast_neg, neg_sq, abs_neg, Int.cast_ofNat, Nat.abs_cast]
+  have : Summable (fun n : ℕ ↦ Real.exp (-n))
```

Comment:
Do you think this is worth being it own lemma? I can imagine it being used again in the future.

Reply:
Moved to `Analysis.SpecialFunctions.Exp`